habitat_sim/robots/mobile_manipulator.py

```diff
@@ -0,0 +1,391 @@
+
+        # update motor settings dict
+        self.joint_motors = {}
+        for motor_id, joint_id in self._robot.existing_joint_motor_ids.items():
+            self.joint_motors[joint_id] = (
+                motor_id,
+                self._robot.get_joint_motor_settings(motor_id),
+            )
+
+    #############################################
+    # ARM RELATED
+    #############################################
+    def get_arm_joint_lims(self) -> Tuple[np.ndarray, np.ndarray]:
+        """Get the arm joint limits in radians"""
+        arm_pos_indices = list(
+            map(lambda x: self.joint_pos_indices[x], self.params.arm_joints)
+        )
+
+        lower_lims = [self.joint_limits[0][i] for i in arm_pos_indices]
+        upper_lims = [self.joint_limits[1][i] for i in arm_pos_indices]
+        return lower_lims, upper_lims
+
+    def get_ee_link_id(self) -> int:
+        """Gets the Habitat Sim link id of the end-effector."""
+        return self.params.ee_link
+
+    def get_ee_local_offset(self) -> mn.Vector3:
+        """Gets the relative offset of the end-effector center from the
+        end-effector link.
+        """
+        return self.params.ee_offset
+
+    def calculate_ee_fk(self, js: np.ndarray) -> np.ndarray:
+        """Gets the end-effector position for the given joint state."""
+        self._robot.joint_positions = js
+        return self.get_end_effector_transform().translation
+
+    def calculate_ee_ik(self, ee_targ: np.ndarray) -> np.ndarray:
+        """Gets the joint states necessary to achieve the desired end-effector
+        configuration.
+        """
+        raise NotImplementedError("Currently no implementation for generic IK.")
+
+    def get_end_effector_transform(self) -> mn.Matrix4:
+        """Gets the transformation of the end-effector location. This is offset
+        from the end-effector link location.
+        """
+        ef_link_transform = self._robot.get_link_scene_node(
+            self.params.ee_link
+        ).transformation
+        ef_link_transform.translation = ef_link_transform.transform_point(
+            self.get_ee_local_offset()
+        )
+        return ef_link_transform
+
+    def set_gripper_state(self, gripper_state: float) -> None:
+        """Set the desired state of the gripper"""
+        # TODO: should this apply the change to motors or joints?
+        self._gripper_state = gripper_state
+
+    def close_gripper(self) -> None:
+        """Set gripper to the close state"""
+        self.set_gripper_state(self.params.gripper_closed_state)
+
+    def open_gripper(self) -> None:
+        """Set gripper to the open state"""
+        self.set_gripper_state(self.params.gripper_open_state)
+
+    def is_gripper_open(self) -> bool:
+        # Give some threshold for the open state.
+        grip_dist = np.abs(self._gripper_state - self.params.gripper_open_state)
+        return grip_dist < 0.001
+
+    def set_arm_pos(self, ctrl: List[float]):
```

Comment:
Maybe make `arm_pos` property as well, with getter and setter.

Reply:
Simulation will change the arm state, so the property may be out of date unless explicitly updated. Maybe better to let the user manage that since we'll otherwise risk a stale state in observation, reward, termination, etc... depending on when update is called. If it ends up becoming inefficient for some reason we can add it later.